wordpress-mobile/WordPress-Android
# Use View Binding in Jetpack feature ViewHolders

This PR introduces view bindings in the ViewHolders in the Jetpack features - Scan, Threats and Backup. I've tried several approaches and in the end I think this one is the best. It still encapsulates the layout inflation to the parent and makes the binding accessible from all the children. It also feels quite safe. Other things I tried:
- make the abstract `onBind` method like this `binding.onBind` - the advantage is that the children doesn't need to ever call the `binding` so it could be private. However, I don't think it makes the code much better or more readable to have the overriden calls look like `JetpackHeaderItemBinding.onBind`
- build the bindings directly in the ViewHolder children - I think passing the inflate function to the parent is safer cause it never happens that the binding is initialized incorrectly (without a parent)

To test:
- Go to a premium Jetpack site (I can give you account details)
- Test the Backup, Threat and Scan features
- Check they look normal

PR submission checklist:
